test/internal/helpers/Player.java

```diff
@@ -2115,9 +2127,32 @@ public static Cleanups withResponseMap(final Map<String, FakeHttpResponse<String
     GenericRequest.resetClient();
     GenericRequest.sessionId = "TEST"; // we fake the client, so "run" the requests
 
-    for (var entry : responseMap.entrySet()) {
-      builder.client.addResponse(entry.getKey(), entry.getValue());
-    }
+    builder.client.setResponseFunc(r -> responseMap.get(r.uri().toString()));
+
+    return new Cleanups(
+        () -> {
+          GenericRequest.sessionId = null;
+          HttpUtilities.setClientBuilder(() -> old);
+          GenericRequest.resetClient();
+        });
+  }
+
+  /**
+   * Sets next response to a GenericRequest Note that this uses its own FakeHttpClientBuilder so
+   * getRequests() will not work on one set separately
+   *
+   * @param responseFunc Function returning responses, keyed by request.
+   * @return Cleans up so this response is not given afterwards.
+   */
+  public static Cleanups withResponseFunc(
```

Comment:
Nitpick is that I would probably call this `withResponse` and allow overloading to handle the fact that we are supplying a function. Could even turn `withResponseMap` into an overload of `withResponse`